src/distrs/wilcox.jl

```diff
@@ -0,0 +1,155 @@
+#=
+The wilcox distribution is equivalent to the problem
+of finding the number of subsets with size nx
+of {1,2,...,nx,nx+1,...,nx+ny} summing to (U-sum(1:nx))
+relative to the total number of subsets with size nx.
+The empty subset is defined to sum to zero.
+This can be calculated using the recursion:
+either nx+ny is in the subset in witch case we need to calculate
+the number of subsets with size nx-1 of {1,2,...,nx,nx+1,...,nx+ny-1}
+summing to (U-sum(1:nx)-nx-ny),
+or nx is not in the subset in which case we need to calculate
+the number of subsets with size nx of {1,2,...,nx,nx+1,...,nx+ny-1}
+summing to (U-sum(1:nx)),
+This can be calculated bottom up using dynamic programming.
+
+The (i,j)'th element of DP in the k'th outer loop iteration represents:
+the number of subsets with size k of {1,2,...,k,k+1,...,k+j-1} summing to i.
+ =#
+
+@inline function wilcoxDP(nx, ny, U)
+    DP = zeros(Int, sum(1:nx)+U, ny + 1)
+    for i in 1:(nx+ny)
+        for j in i:(ny+1)
+            DP[i, j] = 1
+        end
+    end
+    for k in 2:nx
+        for j in 1:(ny+1)
+            for i in min(U + sum(1:k), sum((j):(k+j-1))):-1:1
```

Comment:
I think it should be possible to minimize calculations here a bit:
- We know that `sum(1:k) = (k * (k + 1)) >> 1`, and it could be calculated only once in the outer loop
- Moreover, we know that `sum(j:(k+j-1)) = sum(1:(k+j-1)) - sum(1:(j-1)) = (k * (k + 2 * j - 1)) >> 1`

We even know that `min(U + sum(1:k), sum(j:(k+j-1))) = min(U + sum(1:k), sum(1:k) + k * (j - 1)) = sum(1:k) + min(U, k * (j - 1))`.
Thus we could
a) Precompute `a = (k * (k + 1)) >> 1` in the outer loop (possibly `a` is not a good variable name)
b) Increment `b = b + k` with initial value `b = 0` after every iteration of `for j ...` (could be skipped as soon as `b >= U`)
c) Use
```suggestion
            for i in (a + min(U, b)):-1:1
```
here

Reply:
I took a middle ground between performance and readability.